google/cloud/forseti/services/model/importer/importer.py

```diff
@@ -321,20 +326,26 @@ def model_action_wrapper(session,
         if pre_action:
             pre_action()
 
-        item_counter = 0
-        for inventory_data in inventory_iterable:
-            item_counter = item_counter + 1
+        idx = 0
+        for idx, inventory_data in enumerate(inventory_iterable, start=1):
             if isinstance(inventory_data, tuple):
                 action(*inventory_data)
             else:
                 action(inventory_data)
-            if not item_counter % flush_count:
-                LOGGER.debug('Flushing write session: %s.', item_counter)
+
+            if not idx % flush_count:
+                # Flush database every flush_count resources
+                LOGGER.debug('Flushing write session: %s.', idx)
                 session.flush()
 
         if post_action:
             post_action()
-        return item_counter
+
+        if idx % flush_count:
```

Comment:
This will happen only if the post_action modify the idx variable.

Reply:
No, it happens if the number of items in inventory_iterable is not an even divisor for the flush_count.

For example:

len(inventory_iterable) = 1300
flush_count = 1000

There will be one flush when idx = 1000, and then after the for loop returns, idx will = 1300, which will trigger the second flush for the remain 300 rows.